demisto_sdk/commands/format/update_layout.py

```diff
@@ -255,3 +260,79 @@ def remove_copy_and_dev_suffixes_from_layout(self):
                         section['name'] = remove_copy_and_dev_suffixes_from_str(section.get('name'))
+
+    def remove_inexistent_fields_layoutscontainer(self):
+        """
+        Remove in-existent incident/indicator fields from a container layout.
+        """
+        if not self.id_set_file:
+            return
+
+        content_fields = get_all_incident_and_indicator_fields_from_id_set(self.id_set_file, 'layout')
+        built_in_fields = [field.lower() for field in BUILT_IN_FIELDS] + LAYOUT_AND_MAPPER_BUILT_IN_FIELDS
+
+        layout_container_items = [
+            layout_container_field for layout_container_field in LAYOUT_CONTAINER_FIELDS
+            if self.data.get(layout_container_field)
+        ]
+
+        for layout_container_item in layout_container_items:
+            layout = self.data.get(layout_container_item, {})
+            layout_tabs = layout.get('tabs', [])
+            self.remove_inexistent_fields_from_tabs(
+                layout_tabs=layout_tabs, content_fields=content_fields, built_in_fields=built_in_fields
+            )
+
+    @staticmethod
+    def extract_content_fields(content_fields, built_in_fields):
+        """
+        Get only incident/indicator fields which are part of the id json file.
+        """
+        def _extract_content_fields(field):
+            """
+            Get only incident/indicator fields which are part of the id json file.
+            """
+            field = field.get('fieldId', '').replace('incident_', '').replace('indicator_', '').lower()
+            return field in built_in_fields or field in content_fields
+
+        return _extract_content_fields
+
+    def remove_inexistent_fields_from_tabs(self, layout_tabs, content_fields, built_in_fields):
+        """
+        Remove in-existent fields which are not part of the id json from tabs.
+        """
+        for tab in layout_tabs:
+            layout_sections = tab.get('sections', [])
+            for section in layout_sections:
+                items = section.get('items', [])
+                section['items'] = list(
+                    filter(
+                        self.extract_content_fields(content_fields=content_fields, built_in_fields=built_in_fields),
+                        items
+                    )
+                )
+
+    def remove_inexistent_fields_layout(self):
+        """
+        Remove in-existent fields from a layout.
+        """
+        if not self.id_set_file:
+            return
+
+        content_fields = get_all_incident_and_indicator_fields_from_id_set(self.id_set_file, 'layout')
+        built_in_fields = [field.lower() for field in BUILT_IN_FIELDS] + LAYOUT_AND_MAPPER_BUILT_IN_FIELDS
+
+        layout = self.data.get('layout', {})
+        layout_sections = layout.get('sections', [])
+        for section in layout_sections:
+            fields = section.get('fields', [])
+            section['fields'] = list(
+                filter(
```

Comment:
looks like most of this method cant be replaced the same way, but I think the logic of replace('incident_', '').replace('indicator_') and checking if not in built_in_fields and inc_field not in content_incident_fields
should be possible to share